arch/xtensa/core/userspace.S

```diff
@@ -226,13 +226,12 @@ xtensa_userspace_enter:
 	 */
 	entry a1, 16
 
+	SPILL_ALL_WINDOWS
+
 	/* We have to switch to kernel stack before spill kernel data and
 	 * erase user stack to avoid leak from previous context.
 	 */
 	mov a1, a7 /* stack start (low address) */
-	addi a1, a1, -16
```

Comment:
This used to be reserving 16 bytes of space on the stack, and now it's not.  Is that actually OK?   What was being placed there?  Maybe you want to change the ENTRY instruction above to be a 32 byte frame instead of 16?  Or if it was a mistake, could you explain in the commit message?

Reply:
It is a mistake, I thought we are starting a new context here, A0-A3 spill area is for the caller of the entry function, which doesn't exist. It is like when we are initializing stack, but actually this is needed while we are using the privileged stack because we do callX to erase and set user stack permissions. Will get it back.